frontend/src/components/EditableListItem.vue

```diff
@@ -1,24 +1,53 @@
 <script setup lang="ts">
-import { defineProps } from "vue";
+import { defineProps, onMounted, ref } from "vue";
 
 import DropdownButton from "@/components/DropdownButton.vue";
 import DropdownButtonItem from "@/components/DropdownButtonItem.vue";
-import type { EditableListAction, EditableListItemProps } from "@/components/EditableList.vue";
+import type { EditableListAction, EditableListItemModel } from "@/components/EditableList.vue";
 
-const props = defineProps<{ item: EditableListItemProps; actions?: EditableListAction[] }>();
+const props = defineProps<{ actions?: EditableListAction[] }>();
 
 const emit = defineEmits<{ action: [payload: string] }>();
+
+const item = defineModel<EditableListItemModel>({ required: true });
+const label = ref<HTMLSpanElement>();
+
+function onItemNameEdited(e: Event) {
+  (e.target as HTMLInputElement).blur();
+  item.value.isUnnamed = false;
+  item.value.name = (e.target as HTMLSpanElement).textContent ?? "unnamed";
+}
+
+onMounted(() => {
+  if (item.value.isUnnamed && label.value) {
+    label.value.focus();
+
+    const sel = window.getSelection();
+    if (sel) {
+      const range = document.createRange();
+      range.selectNodeContents(label.value);
+      sel.removeAllRanges();
+      sel.addRange(range);
+    }
+  }
+});
 </script>
 
 <template>
   <div class="editable-list-item">
     <div class="label-container">
-      <span class="icon" v-if="props.item.icon" :class="props.item.icon" />
-      <span class="label">
-        {{ props.item.name }}
+      <span class="icon" v-if="item.icon" :class="item.icon" />
+      <span
+        class="label"
+        :contenteditable="item.isUnnamed"
+        ref="label"
+        @blur="onItemNameEdited"
```

Comment:
What does `blur` do? I can't find a reference to it in the Vue docs

Reply:
It's a native event of inputs (lose of focus).
https://developer.mozilla.org/en-US/docs/Web/API/Element/blur_event